### PR TITLE
Revive the *.csproj solution

### PR DESCRIPTION
**Pull request by bricelam**
With this, `build initialize` will transform the `project.lock.json` files into MSBuild items.

I tried to use NuGet's project.json MSBuild integration, but in the end it was impossible to use the same `project.json` file for both .NET Core CLI and NuGet. The contention was primarily around project-to-project references.

### CODE REVIEW

**Comment by davidfowl** [comment]
Nooooooooo

**Comment by bricelam** [comment]
This was an EF team decision. *.xproj is too painful at the moment. We'll reevaluate again when it stabilizes.

**Comment by Eilon** [comment]
@davidfowl I discussed this with several EF folks, it's critical to unblock progress.

@bricelam I heard there should be new internal builds of tooling this week so hopefully it's not a long wait!

File: src/Microsoft.EntityFrameworkCore.Commands/project.json [hunk 1]
```diff
@@ -12,9 +8,6 @@
     "xmlDoc": true
   },
   "compile": "..\\Shared\\*.cs",
-  "namedResource": {
```
Comment (natemcmaster) [inline src/Microsoft.EntityFrameworkCore.Commands/project.json]: Why don't we need this anymore? Does dotnet-cli automatically include resx files now?

**Comment by natemcmaster** [comment]
:shipit: x 2

File: test/Microsoft.EntityFrameworkCore.SqlServer.FunctionalTests/project.json [hunk 2]
```diff
@@ -8,12 +8,9 @@
     "Microsoft.Extensions.Configuration.Json": "1.0.0-*",
     "Microsoft.Extensions.Configuration.EnvironmentVariables": "1.0.0-*"
   },
-  "content": ["**/*.sql", "config.json"],
-  "publishExclude": [".testPublish/**", "obj/**", "bin/**"],
+  "content": [ "**/*.sql", "config.json" ],
+  "publishExclude": [ ".testPublish/**", "obj/**", "bin/**" ],
```
Comment (bricelam) [inline test/Microsoft.EntityFrameworkCore.SqlServer.FunctionalTests/project.json]: Cool. Will do.

**Pull request merged**